firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLiteRemoteDocumentCache.java

```diff
@@ -57,15 +58,16 @@ public void add(MutableDocument document, SnapshotVersion readTime) {
         !readTime.equals(SnapshotVersion.NONE),
         "Cannot add document to the RemoteDocumentCache with a read time of zero");
 
-    String path = pathForKey(document.getKey());
+    DocumentKey documentKey = document.getKey();
     Timestamp timestamp = readTime.getTimestamp();
     MessageLite message = serializer.encodeMaybeDocument(document);
 
     db.execute(
         "INSERT OR REPLACE INTO remote_documents "
-            + "(path, read_time_seconds, read_time_nanos, contents) "
-            + "VALUES (?, ?, ?, ?)",
-        path,
+            + "(collection_path, document_id, read_time_seconds, read_time_nanos, contents) "
+            + "VALUES (?, ?, ?, ?, ?)",
+        collectionForKey(documentKey),
```

Comment:
These are repeated enough that I think we can add them to `DocumentKey`, like `key.collection()` and `key.documentId()` ?

Reply:
Done